### PR TITLE
WFLY-9613 Further reduce metaspace usage for clustering Externalizers & KeyFormat instances

### PR DESCRIPTION
**Pull request by pferraro**
https://issues.jboss.org/browse/WFLY-9613

### CODE REVIEW

**Comment by bstansberry** [comment]
@pferraro any compatibility concerns here? API or other...

**Comment by pferraro** [comment]
@bstansberry None of the changes affect compatibility or change public API.

**Comment by bstansberry** [comment]
This is ok by me, but Paul's going to have someone go over it for a code review.

File: clustering/marshalling/spi/src/main/java/org/wildfly/clustering/marshalling/spi/Serializer.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+import java.io.IOException;
+
+/**
+ * Serializes an object to/from a binary stream.
+ * @author Paul Ferraro
+ */
+public interface Serializer<T> {
+
+    /**
+     * Writes the specified integer to the specified output stream
```
Comment (rhusar) [inline clustering/marshalling/spi/src/main/java/org/wildfly/clustering/marshalling/spi/Serializer.java]: s/integer/object (in multiple places)

**Review by rhusar — APPROVED** [review]

**Pull request merged**